### PR TITLE
Only include levelbuilder block preview on Blockly levels

### PR DESCRIPTION
**Pull request by joshlory**
Fixes an exception when creating Game Lab levels on Level Builder.

image

### CODE REVIEW

**Comment by caleybrock** [comment]
This makes sense to me. Looks like it would work for applab too since it's not blockly.

File: dashboard/app/views/levels/editors/_blockly.html.haml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 - content_for(:head) do
-  = javascript_include_tag 'js/blockly'
-  - if @level.game && @level.game.app
+  - if @level.is_a?(Blockly) && !@level.game.uses_droplet?
+    = javascript_include_tag 'js/blockly'
     = javascript_include_tag "js/en_us/blockly_locale"
     = javascript_include_tag "js/en_us/common_locale"
     = javascript_include_tag "js/en_us/#{@level.game.app}_locale"
```
Comment (Hamms) [inline dashboard/app/views/levels/editors/_blockly.html.haml:6]: can a level `be_a?(Blockly)` without having a `game` that also has an `app`? If so, we'll still need that check. If not, this LGTM! Thanks for cleaning up my indiscretion.

Comment (joshlory) [reply dashboard/app/views/levels/editors/_blockly.html.haml:6]: All have `game.app` for me locally:

``` ruby
irb(main):001:0> Blockly.all.all?{ |l| l.game.app }
=> true
```

FWIW it looks like the only `Level` objects that don't have an associated `Game` are `FrequencyAnalysis` levels.  I wonder if we should fix these and make `game_id` a non-nullable column?

``` ruby
irb(main):002:0> Level.where(game_id: nil).pluck(:type).uniq
=> ["FrequencyAnalysis"]
```

Comment (Hamms) [reply dashboard/app/views/levels/editors/_blockly.html.haml:6]: odd that that's true for `FrequencyAnalysis` but not `Vigenere`. I think that'd be worthwhile.

**Pull request merged**